crates/proof-of-sql/tests/timestamp_integration_tests.rs

```diff
@@ -134,6 +134,86 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_basic_timestamp_inequality_query() {
+        let test_timestamps = vec![i64::MIN, -1, 0, 1, i64::MAX];
+
+        run_timestamp_query_test(
+            "SELECT * FROM table WHERE times < timestamp '1970-01-01T00:00:00Z';",
+            test_timestamps.clone(),
+            vec![i64::MIN, -1],
+        );
+
+        run_timestamp_query_test(
+            "SELECT * FROM table WHERE times > timestamp '1970-01-01T00:00:00Z';",
+            test_timestamps.clone(),
+            vec![1, i64::MAX],
+        );
+
+        run_timestamp_query_test(
+            "SELECT * FROM table WHERE times >= timestamp '1970-01-01T00:00:00Z';",
+            test_timestamps.clone(),
+            vec![0, 1, i64::MAX],
+        );
+
+        run_timestamp_query_test(
+            "SELECT * FROM table WHERE times <= timestamp '1970-01-01T00:00:00Z';",
+            test_timestamps.clone(),
+            vec![i64::MIN, -1, 0],
+        );
+    }
+
+    #[test]
+    fn test_timestamp_inequality_queries_with_timezone_offsets() {
+        // Test with a range of timestamps around the Unix epoch
+        // 60 * 60 = 3600 * 8 (PST offset) = 28800
+        let test_timestamps = vec![i64::MIN, -28800, -28799, 0, 1, i64::MAX];
```

Comment:
Maybe you want to use 28800 and 28799 instead so that the filter actually tells the difference between values?